earthfile2llb/interpreter.go

```diff
@@ -1040,6 +1041,9 @@ func parseArgArgs(ctx context.Context, cmd spec.Command) (argOpts, string, *stri
 		if args[1] != "=" {
 			return argOpts{}, "", nil, errInvalidSyntax
 		}
+		if variables.IsBuiltIn(args[0]) {
+			return argOpts{}, "", nil, errBuiltInArgHasDefault
+		}
```

Comment:
I would put this check in `variables.ParseArgs` rather than in the interpreter, because ParseArgs is already called from this common point: https://github.com/earthly/earthly/blob/main/earthfile2llb/converter.go#L1296, that is used whenever another target is referenced (FROM, COPY, BUILD, WITH DOCKER --load, etc).

Reply:
The other relevant place to add the check to is `variables.ParseCommandLineArgs` (for the args passed via the command-line). This is called from here: https://github.com/earthly/earthly/blob/main/cmd/earthly/main.go#L2693.

Reply:
Oh, I just realized, I think we'd also need to do a check in `variables.DeclareArg` (called from here: https://github.com/earthly/earthly/blob/main/earthfile2llb/converter.go#L1057) so that the user cannot assign a default value (which would be ignored) to a builtin variable.